exonum/src/runtime/rust/interfaces.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Important interservice communication interfaces.
+
+use crate::{
+    merkledb::BinaryValue,
+    runtime::{DispatcherError, MethodId},
+};
+
+use super::{ExecutionError, Interface, TransactionContext};
+
+/// Identifier of the [`Initialize::initialize`] method.
+///
+/// [`Initialize::initialize`]: trait.Initialize.html#tymethod.initialize
+pub const INITIALIZE_METHOD_ID: MethodId = 0;
+/// Fully qualified name of the `Initialize` interface.
+pub const INITIALIZE_INTERFACE_NAME: &str = "Initialize";
+
+/// This trait describes a service interface to pass initial configuration parameters into
+/// the started service instance.
+///
+/// This interface is optional, therefore if initialize was called on a service
+/// that does not implement this interface, and then if the configuration parameters are empty,
```

Comment:
The initialization is successful if both statements are valid:
1. Service does not implement this interface
2. Configuration parameters are empty
? 
If so, why can't I initialize a service without any parameters?